dynamodb/src/main/scala/zio/dynamodb/DynamoDBQuery.scala

```diff
@@ -181,6 +181,14 @@ sealed trait DynamoDBQuery[+A] { self =>
       case _                          => self
     }
 
+  def parallel(n: Int): DynamoDBQuery[A] =
```

Comment:
we probably need a small amount of ScalaDoc here or somewhere else to explain that there are no ordering guarantees of the items returned in the stream when the query is executed due to the parallel processing and the and the lack ordering guarantees offered by `ZStream.merge`. 

For the first release I don't think this is a big issue. Also even it you were coding this by hand using the SDK there would be no global ordering guarantees due to the parallel processing.

If ordering within a segmented partition proves to be important, then later on we could offer something like a new query that returns a tuple or a collection of streams that user can index on partition.

Reply:
One thing I discussed with John today was around what to do if a user passes in an `n < 0`. He suggested mirroring some of ZIO's failure cases like `failCause` or `dieCause`. I'm not sure if that's something we want to do now or look into in the future? Right now this less than zero case is covered.

Reply:
how about we aim to functionality completeness for the 1st (shapshot enabled) release, gain some feedback and circle back to this one? It sounds like a good idea and we should create an issue for it.